src/solver/page/request-drawer/SolverLimitationsDrawer.tsx

```diff
@@ -1,84 +1,167 @@
-                      width={24}
-                      height={24}
-                    />
-                    {building.name}
+                    <FactoryItemImage size={24} id={resource} />
+                    {item.name}
                   </Group>
                 }
-                checked={!request?.blockedBuildings?.includes(building.id)}
+                checked={!request?.blockedResources?.includes(resource)}
                 onChange={e =>
                   useStore
                     .getState()
-                    .toggleBlockedBuilding(
+                    .toggleBlockedResource(
                       id!,
-                      building.id,
+                      resource,
                       !e.currentTarget.checked,
                     )
                 }
               />
-            ))}
-          </Stack>
-        </SimpleGrid>
-      </Stack>
-    </>
+            );
+          })}
+        </Stack>
+        <Stack gap="xs">
+          <Text size="lg">Buildings</Text>
+          {FactoryBuildingsForRecipes.map(building => (
+            <Checkbox
+              key={building.id}
+              label={
+                <Group gap="xs">
+                  <Image
+                    src={building.imagePath.replace('_256', '_64')}
+                    width={24}
+                    height={24}
+                  />
+                  {building.name}
+                </Group>
+              }
+              checked={!request?.blockedBuildings?.includes(building.id)}
+              onChange={e =>
+                useStore
+                  .getState()
+                  .toggleBlockedBuilding(
+                    id!,
+                    building.id,
+                    !e.currentTarget.checked,
+                  )
+              }
+            />
+          ))}
+        </Stack>
+        <Stack gap="sm" style={{ gridColumn: 'span 2' }}>
+          <Text size="lg">Logistics</Text>
+          <Text size="sm">
+            Selecting a belt or a pipeline will highlight in the calculator all
+            the ingredients which are exceeding the belt / pipeline amount.
+          </Text>
+        </Stack>
+        <Stack gap="xs">
+          <Text size="lg">Belt Logistic</Text>
+          {FactoryConveyorBelts.map((building, index) => (
+            <Radio
+              key={building.id}
+              label={
+                <Group gap="xs">
+                  <Image
+                    src={building.imagePath.replace('_256', '_64')}
+                    width={24}
+                    height={24}
+                  />
+                  {building.name}
+                </Group>
+              }
+              checked={
+                maxBelt === building.id ||
+                (!maxBelt && index === FactoryConveyorBelts.length - 1)
+              }
+              onChange={e =>
+                useStore.getState().updateGameSettings(state => {
+                  setByPath(state, 'maxBelt', building.id);
```

Comment:
here since you already know the key you can go with `settings => settings.maxBelt = building.id;` which si cleaner

Reply:
I just checked the code and don't really know what you mean

Reply:
nvm.